bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -141,17 +142,37 @@ final class BagsManager(executorService: ExecutorService) {
 
         val key = obs.getNodeKey
         state += key
-        Future {
-          // If dequeue is false this means that (a) another task scheduled *after* me ended up
-          // running before me, so their result is as good as mine would have been and we're done;
-          // or (b) we don't care about that program anymore, so we're done.
-          if (dequeue(key, obs.getProgramID)) {
-            // Otherwise construct an obs context, verify that it's bags-worthy, and go.
-            ObsContext.create(obs).asScalaOpt.foreach { ctx =>
-              val eligibleForBags = isEligibleForBags(ctx)
-              if (eligibleForBags
-                && hasBeenUpdated(obs, ctx)
-                && notObserved(obs)) {
+
+        def status(newStatus: BagsStatus): Unit = {
+          val oldStatusOpt = state.statuses.get(key)
+          if (!oldStatusOpt.exists(_.equals(newStatus))) {
+            state = state.setStatus(key, newStatus)
+            notifyBagsStatusListeners(observation, oldStatusOpt, Some(newStatus))
+          }
+        }
+        def clearStatus(): Unit = {
+          state.statuses.get(key).foreach(oldStatus => {
+            state = state.clearStatus(key)
+            notifyBagsStatusListeners(observation, Some(oldStatus), None)
+          })
+        }
+
+
+        // If dequeue is false this means that (a) another task scheduled *after* me ended up
+        // running before me, so their result is as good as mine would have been and we're done;
+        // or (b) we don't care about that program anymore, so we're done.
+        clearStatus()
+
+        if (dequeue(key, obs.getProgramID)) {
```

Comment:
I wanted to set the `BagsStatus` to `Pending` only if the observation could be dequeued, and before the `Future` was executed. Does that make sense? I was a bit wary of this, but I wasn't sure a better way to do it offhand.